sklearn/metrics/ranking.py

```diff
@@ -347,13 +368,88 @@ def _binary_roc_auc_score(y_true, y_score, sample_weight=None):
         return 0.5 * (1 + (partial_auc - min_area) / (max_area - min_area))
 
     y_type = type_of_target(y_true)
-    if y_type == "binary":
+    y_true = check_array(y_true, ensure_2d=False, dtype=None)
+    y_score = check_array(y_score, ensure_2d=False)
+
+    if y_type == "multiclass" or (y_type == "binary" and
+                                  y_score.ndim == 2 and
+                                  y_score.shape[1] > 2):
+        # validation of the input y_score
+        if not np.allclose(1, y_score.sum(axis=1)):
+            raise ValueError(
+                "Target scores need to be probabilities for multiclass "
+                "roc_auc, i.e. they should sum up to 1.0 over classes.")
+
+        # do not support partial ROC computation for multiclass
+        if max_fpr is not None and max_fpr != 1.:
+            raise ValueError("Partial AUC computation not available in "
+                             "multiclass setting. Parameter 'max_fpr' must be"
+                             " set to `None`. Received `max_fpr={0}` "
+                             "instead.".format(max_fpr))
+
+        # validation for multiclass parameter specifications
+        average_options = ("macro", "weighted")
+        if average not in average_options:
+            raise ValueError("Parameter 'average' must be one of {0} for"
+                             " multiclass problems.".format(average_options))
+        multiclass_options = ("ovo", "ovr")
+        if multiclass not in multiclass_options:
+            raise ValueError("Parameter multiclass='{0}' is not supported"
+                             " for multiclass ROC AUC. 'multiclass' must be"
+                             " one of {1}.".format(
+                                 multiclass, multiclass_options))
+        if labels is not None:
+            unique_labels = np.unique(labels)
+            if len(unique_labels) != len(labels):
+                raise ValueError("Parameter 'labels' must be unique")
+            if len(unique_labels) != y_score.shape[1]:
+                raise ValueError(
+                    "Number of given labels not equal to the number of "
+                    "columns in 'y_score'")
+            if set(np.unique(y_true)) > set(unique_labels):
+                raise ValueError(
+                    "'y_true' contains labels not in parameter 'labels'")
+        if multiclass == "ovo":
+            if sample_weight is not None:
+                raise ValueError("Parameter 'sample_weight' is not supported"
+                                 " for multiclass one-vs-one ROC AUC."
+                                 " 'sample_weight' must be None in this case.")
+            if labels is not None:
+                y_true_multiclass = np.empty_like(y_true, dtype=np.int32)
+                for i, label in enumerate(labels):
+                    y_true_multiclass[y_true == label] = i
```

Comment:
You could be using a label binarizer here as well right?

Reply:
`LabelBinarizer` would not work for this use case, since it returns a 2 dimensional array. `LabelEncoder` or `sklearn.preprocessing.label._encode` would have worked, if it did not assume that `uniques` is ordered.

In this context, the `labels` may be an unordered list, where the positions represents different columns in `y_score`